### PR TITLE
Switchboard

### PR DESCRIPTION
**Pull request by Natim**
When I run `npm run start` I've got this error:

```
Module parse failed: ~/switchboard-experiments-kinto/switchboard/src/index.js Unexpected token (19:2)
You may need an appropriate loader to handle this file type.
SyntaxError: Unexpected token (19:2)
    at Parser.pp.raise (~/switchboard-experiments-kinto/switchboard/node_modules/acorn/dist/acorn.js:920:13)
```

### CODE REVIEW

**Comment by Natim** [comment]
I did a first ReactJS app that uses the initialState to display the app layout.

Could @n1k0 give me an early feedback on my component architecture?

**Comment by Natim** [comment]
My next step is load the state dynamically.

File: switchboard/src/ab_testing.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+export function matches(feature, userAgent, languages, country) {
+  // Handle Match
+  const match = feature.match;
+  let matches = true;
+
+  // User-Agent match
+  if (match.hasOwnProperty('userAgent')) {
+    const uaMatching = new RegExp(match.userAgent).exec(userAgent);
+    if (!uaMatching)  matches = false;
```
Comment (n1k0) [inline switchboard/src/ab_testing.js]: Nit: `matches = new RegExp(match.userAgent).test(userAgent);`

File: switchboard/src/ab_testing.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  // Language Matching
+  if (match.hasOwnProperty('lang')) {
+    const featureLanguages = languages.filter(language => {
+      return new RegExp(match.lang).exec(language);
+    });
+
+    if (languages.length === 0) {
+      matches = false;
+    }
```
Comment (n1k0) [inline switchboard/src/ab_testing.js]: Nit: 

``` js
    matches = languages.some(language => new RegExp(match.lang).test(language));
```

File: switchboard/src/ab_testing.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  // Country Matching
+  if (match.hasOwnProperty('country')) {
+    if (!country.match(match.country)) {
```
Comment (n1k0) [inline switchboard/src/ab_testing.js]: hmm you sure?

File: switchboard/src/ab_testing.js (hunk 1 above)
Comment (Natim) [inline switchboard/src/ab_testing.js]: The problem is I want to change matches only if it is false. Not if it is true.

File: switchboard/src/ab_testing.js (hunk 3 above)
Comment (Natim) [inline switchboard/src/ab_testing.js]: Yes you are right a new RegExp(match.country) should be there.

**Comment by Natim** [comment]
@n1k0 I have added my sagas, it works but the spinner seems to be rendered after the sagas got forked.

**Comment by Natim** [comment]
englishbanner
frenchbanner

**Comment by n1k0** [comment]
Sweet!

> I have added my sagas, it works but the spinner seems to be rendered after the sagas got forked.

The app loads too fast for me so I can't even admire a spinner at all. From what I read in the code the flow looks good though.

Note: you may want to try setting up https://github.com/gaearon/redux-devtools to inspect the actions flow more easily.

**Comment by Natim** [comment]
Here we go :)

**Pull request merged**